### DISCUSSION TITLE
Can I keep my database file in the open Internet?

### DISCUSSION THREAD

**Discussion by jonatasbaldin** [keepassxreboot/keepassxc#6898] [Q&A] [answered]
Hi there 👋 

First, I'd like to thanks all the maintainers and contributors for KeePassXC, I've migrating from 1Password lately in a very smoothly experience.

So far, I've setup my local machine (OSX) and I'm on the mission to setup my phone (iOS).

[Strongbox Safe](https://strongboxsafe.com/) has support for KeePass. One could add the database file from different methods, like Dropbox, Google Drive, WebDAV, SFTP, URL etc.

With that, I was wondering:
- How safe it is to put my database file out in the wild (assuming I'm using a very strong passphrase but not key file)?
- If I use a key file (manually set it in my machine/phone), would it be safer to expose the database file online?
- How long one would take to brute force my file with a strong pass phrase?
- Are there any history of zero-day vulnerabilities to break into the database file? Are those common?

Thanks!

**Comment by yi0n** [comment]
Hi,

sorry for capturing your thread, but I want to add my thoughts/ideas. Also possible dangers on uploading the database to the internet.

___First: I would find it very useful to have virtual identities. Because a name is always connected to different entries.
Example: Person John Hammond. He has the email addresses j.hammond67@gmail.com and john.hammond@myhome.dyndns.net. Also a lot of other entries are connected with these email addresses. Like forum accounts, etc. pp.
A way of handling this is by using an extra keepass database for each identity. But it would also be nice to handle all in a master db.
Especially for managing club/community databases.

___Second: A way of handling a KeePass database in the open Internet, it would be nice, **not** to unlock the whole database at once or having the complete db outsourced. Who knows how they might be duplicated/copied on cloud services or a algorithm/cipher can be broken in some years in the future.
Instead, something like this KeeShare approach only with one-time-tokens.
What's in my mind is something like this:
1. My main KeePass database is kept on my computer at home
2. I sync an extra KeeShare database (which isn't able to save back into my main) to a cloud service or different devices (for access without internet)
3. On that different devices, I connect to my db on the cloud storage or on foreign computers I could access a KeePass (REST like) microservice from my cloud storage like /{user}/awesomeforum.org
4. KeePass generates a token and an auth-digit-code that is sent to my phone/email
5. I legitimate that I've requested the password
6. KeePass gives me the password
(That could be even driven further if websites offer some API for password changes and KeePass can generate each time on each request a new password or OTP for a limited lifespan)

Just some ideas...

Regards

**Comment by droidmonkey** [comment, marked as answer]
The "Open Internet" is very different from personal cloud storage like Dropbox and OneDrive. Storing the database in your personal accounts is very safe and encouraged. The file is _always encrypted_ and secrets will never be exposed to those services. A strong database password is all you need, 15+ characters is more than sufficient. Due to the use of key derivation functions, cracking a database is impossible with a strong password.

**Comment by jonatasbaldin** [comment]
Thanks @droidmonkey.

In my case, I don't want to use services like Dropbox and OneDrive, I'd like to be able to have my database availability as simple as possible, like in a simple URL.

> Due to the use of key derivation functions, cracking a database is impossible with a strong password.

With that in mind, if anyone puts the hand on my database, they wouldn't be able to crack it anyway, right?

**Reply by droidmonkey** [reply]
They'd have to have about 100 billion years of computing power for a 20 character randomized password.

**Reply by jonatasbaldin** [reply]
yep, that will do it hahaha thanks again!
